stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/WebSocketDispatcher.java

```diff
@@ -205,6 +214,45 @@ public void notifyFailure(BStruct error) {
         Executor.submit(onCloseResource, onCloseCallback, null, null, bValues);
     }
 
+    public static void dispatchError(WebSocketOpenConnectionInfo connectionInfo, Throwable throwable) {
+        WebSocketService webSocketService = connectionInfo.getService();
+        Resource onErrorResource = webSocketService.getResourceByName(WebSocketConstants.RESOURCE_NAME_ON_ERROR);
+        if (isUnexpectedError(throwable)) {
```

Comment:
Shouldn't we ideally pass the errors related to WebSocket connection in onError resource. IMO all other internal errors should be logged in internal log but should not appear in onError resource. 
WDYT?

Reply:
@riyafa and all, 
After a discussion, it seems that it is better to indicate every error since there are situations that ballerina user has to get post decisions depending on any error. Hense changed the code to support any error.
(eg: Close other associated connection if any error occurred in a particular connction)